### PR TITLE
Duplicate Restaurant insertion into DB Bug Fix - Verified

### PR DESCRIPTION
**Pull request by nj808**
- added some async .then() chaining to the searchRestaurants() function 
-  Tested the bug fix on a fresh, duplicate database, and the fix seems to have resolved the duplicate restaurant bug

### CODE REVIEW

**Review by gbashour — CHANGES_REQUESTED** [review]
A couple of general comments here:
- Calls to the DB are re-used code. `/getRestaurant` and `/createRestaurant` for example are 2 endpoints that provide the same functionality but are unused. We should be moving functions out of these nested loops and breaking the code into smaller, more modular sections to make it more manageable.
- To this point, utilisation of `try-catch` for these calls helps us have better error handling.
- I think we may need some more rigorous testing with a clear DB to confirm this is an actual fix given the bug wasn't able to be replicated.

File: server/routes/restaurant.js [hunk 1]
```diff
@@ -32,146 +32,146 @@ router.get('/searchRestaurants/:latitude/:longitude/:radius', async (req, res) =
...
+                    } 
+                });
             }
-        }  
-        //a wierd bug where the restaurant_ids array were returning duplicates, quick fix for now
-        res.json(Array.from(new Set(restaurant_ids)));
- 
+        });
+            //a wierd bug where the restaurant_ids array were returning duplicates, quick fix for now
+            res.json(Array.from(new Set(restaurant_ids)));
```
Comment (gbashour) [inline server/routes/restaurant.js]: I think we need to take a closer look at this. We've got a comment that already tells us the `restaurant_ids` array were returning duplicates, yet we only filter these out post calling the `testFlow`.

**Comment by nj808** [comment]
Hey,
- I'm trying to test if the bug fix works using a duplicate MongoDB database 
- I can amend the searchRestaurants call to call create restaurant()

File: server/routes/restaurant.js (hunk 1 above)
Comment (nj808) [reply server/routes/restaurant.js]: Hey,
- I'm trying to test if the bug fix works using a duplicate MongoDB database 
- I can amend the searchRestaurants call to call create restaurant()

Comment (gbashour) [reply server/routes/restaurant.js]: Sounds good! Currently also testing with @abrarbarbhuyia - what we've done is we use your `Set()` fix but call it prior to the for loop, fingers crossed!🤞

Comment (nj808) [reply server/routes/restaurant.js]: Oh, I totally forgot about this bug, I think it is caused by the same error that is duplicating restaurants

Comment (nj808) [reply server/routes/restaurant.js]: I tried to address the review changes you requested in the latest commit - added try catch {} to catch errors and reused the /createRestaurant and /getRestaurant in searchRestaurant() call

**Comment by nj808** [comment]
Well the fix doesn't work, so I have decided to close the pull request

**Pull request closed without merging**